src/ui/public/modules.js

```diff
@@ -1,9 +1,43 @@
+/**
+ * This module is used by Kibana to create and reuse angular modules. Angular modules
+ * can only be created once and need to have their dependencies at creation. This is
+ * hard/impossible to do in require.js since all of the dependencies for a module are
+ * loaded before it is.
+ *
```

Comment:
The angular module's dependencies don't actually need to be available at creation time. Consider this here:
---
app.js
```js
angular.module('app', ['ui.bootstrap'])
.controller('AppController', function () { ... });

require('./directive');
```

---
directive.js
```js
angular.module('app')
.directive('someDirective', function () { ... });
```

---

In this scenario require.js would load directive.js first, which would then cause angular to through an error because the "app" module does not exist yet. This effect is increased when app.js links off to many different modules.

This is normally solved by creating unique modules per file, listed as the 1st alternate solution below. Unfortunately this solution would have required that we replicate our require statements, but perhaps that is how we should have handled it.